frontend/homework/8/code.ts

```diff
@@ -0,0 +1,65 @@
+interface Recipe {
+    image: string;
+    name: string;
+    rating: number;
+    cuisine: string;
+    difficulty: string;
+    ingredients: string[];
+    timeTaken: number;
+    calorieCount: number;
+}
+
+async function fetchRecipesFromAPI(): Promise<Recipe[]> {
+    try {
+        const response = await fetch("https://dummyjson.com/recipes");
+        const data = await response.json();
+        const recipes: Recipe[] = data.recipes.map((item: any) => ({
+            image: item.image,
+            name: item.name,
+            rating: item.rating,
+            cuisine: item.cuisine,
+            ingredients: item.ingredients,
+            difficulty: item.difficulty,
+            timeTaken: item.prepTimeMinutes + item.cookTimeMinutes,
+            calorieCount: item.caloriesPerServing
```

Comment:
This part of code can be wrapped in a util function and be reused in both functions